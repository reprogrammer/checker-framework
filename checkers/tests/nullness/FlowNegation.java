public class FlowNegation {

    void testSimpleValid() {
        String s = "m";
        s.toString();
    }

    void testCase1() {
        String s = "m";
        if (s != null) { }
        else { }
        s.toString();
    }

    void testCase2() {
        String s = "m";
        if (s != null) { }
        else { }
        s.toString();
    }
    
    void testInvalidCase1() {
        String s = "m";
        if (s != null) { s = null; }
        else { }
        s.toString();   // error
    }

    void testInvalidCase2() {
        String s = "m";
        if (s != null) { }
        else { s = null; }
        s.toString();   // error
    }
}
