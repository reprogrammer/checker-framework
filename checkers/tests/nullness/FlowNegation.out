:26: (dereference.of.nullable)
:33: (dereference.of.nullable)
:56: (dereference.of.nullable)
:62: (dereference.of.nullable)
